src/constants.js

```diff
@@ -59,6 +60,17 @@ export const CONFIG_DIRECTORY = getDirectory('config');
 export const LINK_REGISTRY_DIRECTORY = path.join(CONFIG_DIRECTORY, 'link');
 export const GLOBAL_MODULE_DIRECTORY = path.join(CONFIG_DIRECTORY, 'global');
 
+export const NODE_BIN_PATH = process.execPath;
+export const YARN_BIN_PATH = getYarnBinPath();
+
+function getYarnBinPath(): string {
+  if (isWebpackBundle) {
+    return __filename;
+  } else {
+    return path.join(__dirname, '..', 'bin', 'yarn.js');
```

Comment:
There's not a lot of option otherwise, there's no reliable way of automatically getting the entry point to yarn. You have to manually handle each case